src/basic/mountpoint-util.c

```diff
@@ -359,6 +359,7 @@ bool fstype_can_uid_gid(const char *fstype) {
 
         return STR_IN_SET(fstype,
                           "adfs",
+                          "exfat",
```

Comment:
is that an upstream fs? i.e. in the regular kernel?

Reply:
There are 2 implementations:

- exfat-fuse https://github.com/relan/exfat
- exfat-nofuse https://github.com/dorimanx/exfat-nofuse

Both the FUSE-based and kernel module support the uid and gid options. The exFAT filesystem spec does not include file ownership or permissions.

Reply:
To more directly answer the question: no, there is no implementation in the mainline kernel due to license and/or patent issues.

Reply:
Is there any perspective for this to go upstream? I through microsoft donated the relevant patents to OIN?

I am a bit concerned about adding these entries for stuff that has no perspective to ever going upstream and being officially supported upstream...

Reply:
> Is there any perspective for this to go upstream? I through microsoft donated the relevant patents to OIN?

I am not sure.

> I am a bit concerned about adding these entries for stuff that has no perspective to ever going upstream and being officially supported upstream...

I don't really understand this concern. I don't think exfat is going to disappear anytime soon.

The existing FUSE implementation is supported on several major distros, including Debian and Fedora.

In the worst, very unlikely case, a third exfat implementation might be merged into the mainline kernel without supporting the uid/gid options. I just can't see that happening.

Reply:
Yeah, I think it's OK to put an established-even-if-not-upstream filesystem here. The maintenance "cost" is negligible on our side, but it makes integration with such file systems much nicer for the users.